### DISCUSSION TITLE
Multiple owners for Vavr and where is the roadmap?

### DISCUSSION THREAD

**Discussion by diversit** [vavr-io/vavr#2929] [General]
Hi,

@pivovarit Thank you for taking up the new ownership of Vavr. It's great this project can continue.
Are you open to have multiple owners on this project?
Having multiple owners would give this project more reliability and continuity as
- the project will not have the same problem again if an owner wants to quit
- to have multiple people talk, discuss and decide about the roadmap
- to have someone to review the owner

For the 10.5 release I saw that @pivovarit had made several changes and approved and merged his own code. This is an issue for this project to be reliable and trust worthy. I assume we don't want to make this project into another 'XZ Utils backdoor' so a proper approval process is required and approving of own PR's should not be allowed for this project to gain more users and supporters.

I was happy to see that during this year multiple people stood up and volunteered to take ownership of the project to secure its continuation. Finding more help should be possible. I myself would also be interested in contributing to this project.

Where is the new roadmap for Vavr? The 10.5 release says there now is a roadmap, but I am unable to find it.

Cheers,
Joost

**Comment by diversit** [comment]
@pivovarit A bit disappointed you have not reacted to this discussion yet. To be fair, this does not give me a good feeling about the future of and course for Vavr.

**Comment by pivovarit** [comment]
> Are you open to have multiple owners on this project?

I am - in fact, Vavr already has a concept of a _core team_ and I hope I will manage to reignite the spark within it. My ultimate goal is to end up with a setup where it would be impossible for me to stop the project even if I really wanted to. While this might be hard to achieve in practice, I believe it's a good stretch goal. It would be hard to reach because an organization admin has ultimate power and can bypass everything - that's true for every project hosted on GitHub.

While I technically possess the power to push changes directly to the main branch, I still issue Pull Requests to ensure visibility and create opportunity for other people to comment. Release job runs are also public and auditable, which was not the case in the past.

> I myself would also be interested in contributing to this project.

I'm awaiting your contributions then. For example, I'm looking for extra helping hands here: https://github.com/vavr-io/vavr/issues/2906

> Where is the new roadmap for Vavr? The 10.5 release says there now is a roadmap, but I am unable to find it.

Multiple answers were posted in at least a couple of issues here - the official discussion with my proposals will be opened this month. 

> A bit disappointed you have not reacted to this discussion yet. To be fair, this does not give me a good feeling about the future of and course for Vavr.

Well, I'm sorry to hear that.

GitHub has this great feature that automatically notifies everyone about maintainers' limited availability, and I happened to have it set since October 28th this year:

<img width="468" alt="status" src="https://github.com/user-attachments/assets/badcf637-41a4-46b4-bf3d-e6800290de69" />

Despite this, you decided to make embittered comments anyway - I want to make sure that readers are aware of that.

A gentle reminder that this project is free and is maintained by volunteers who dedicate their free time and energy to keeping it alive. While feedback is appreciated, I’d like to kindly remind you that open-source projects, that are not backed by companies, operate on a goodwill basis, which has some natural limitations.

**Reply by diversit** [reply]
Thanks your response.
I was not aware of the availability feature in Github, nor did I see it. So sorry for that.
I'm also sorry you feel my comments are 'embittered'. That was not intended. I was just concerned. I appreciate you stepping up as the new maintainer giving this project a new future. Great to hear there already is a 'core' team.

I do appreciate open-source projects and try to help out where I can. Not only by reporting issues but also by providing solutions. I plan on having a look at Vavr features where help is wanted.

I have talked to several Oracle people this year about adding a new collections api to Java, but that is just not going to happen. So for me Vavr is a crucial project helping me creating better applications and I will be more than happy to prove a helping hand.

Looking forward to new features and releases of Vavr.

**Comment by pivovarit** [comment]
You can find the roadmap here: https://github.com/orgs/vavr-io/discussions/2953
